velox/exec/GroupingSet.h

```diff
@@ -91,6 +91,18 @@ class GroupingSet {
     return table_ ? table_->numDistinct() : 0;
   }
 
+  /// Returns number of global grouping sets.
+  vector_size_t numGlobalGroupingSets() const {
```

Comment:
I wonder if we could combine numGlobalGroupingSets and hasDefaultGlobalGroupingSetOutput methods into a single method that returns std::optional<vector_size_t> or just vector_size_t with the number of rows to be produced. Zero or std::nullopt will indicate that no rows need to be produced.